src/containers/ExperimentContainer.tsx

```diff
@@ -68,31 +74,66 @@ export const ExperimentContainer = () => {
   // Get the experiment object and the index of the current module.
   const dispatch = useDispatch()
   const experiment = useSelector(experimentSelector)
+  const experimentModules = useSelector(allModulesSelector)
   const currentModule = useSelector(currentModuleSelector)
 
   // If the user has been 'screened out' then show respective screen
   if (experiment.participantRejected) {
     return <RejectionScreen onExit={() => terminateExperiment(false)} />
   }
 
+  // If all modules have been completed...
+  if (experiment.currentModuleIndex === experiment.definition?.modules.length) {
+    // Mark experiment complete
+    if (!experiment.isComplete)
+      dispatch(
```

Comment:
Correct me if I'm wrong but I don't think we should be updating state directly in the render code path. We should wrap this whole thing in a `useEffect`.

Reply:
It should be fine? Moving some similar code in a previous MR to a `useEffect` just caused UI bugs that were picked up in acceptance?